src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/HttpExtensions.cs

```diff
@@ -37,5 +38,25 @@ public static SummaryType GetSummaryType(this HttpContext context, ILogger logge
 
             return SummaryType.False;
         }
+
+        public static bool GetIsPretty(this HttpContext context, ILogger logger)
+        {
+            var query = context.Request.Query[KnownQueryParameterNames.Pretty].FirstOrDefault();
+
+            if (!string.IsNullOrWhiteSpace(query))
+            {
+                if (!bool.TryParse(query, out bool isPretty))
+                {
+                    // Assume no pretty formatting if parameter can't be parsed.
+                    isPretty = default;
+                }
+
+                logger.LogDebug("Changing response pretty formatting to '{0}'", isPretty);
```

Comment:
I would not log here. This get method would ideally be free of side-effects (and you are not "changing" anything here).

Reply:
Makes sense, thanks. Removed in 827c3e5.